src/Raven.Client/Http/DefaultRavenHttpClientFactory.cs

```diff
@@ -0,0 +1,178 @@
+    {
+        if (GlobalHttpClientCache.TryGetValue(key, out var value))
+            return value.Value.HttpClient;
+
+        value = GlobalConcurrentHttpClientCache.GetOrAdd(key, cacheKey => new Lazy<HttpClientCacheItem>(() => new HttpClientCacheItem(CreateClient(cacheKey, createHttpClient), SystemTime.UtcNow)));
+
+        GlobalHttpClientCache = new Dictionary<HttpClientCacheKey, Lazy<HttpClientCacheItem>>(GlobalConcurrentHttpClientCache);
+
+        return value.Value.HttpClient;
+    }
+
+    public bool TryRemoveHttpClient(HttpClientCacheKey key, bool force = false)
+    {
+        if (GlobalConcurrentHttpClientCache.TryGetValue(key, out var client) &&
+            ((client.IsValueCreated &&
+              SystemTime.UtcNow - client.Value.CreatedAt > MinHttpClientLifetime) || force))
+        {
+            if (GlobalConcurrentHttpClientCache.TryRemove(key, out _))
+                GlobalHttpClientCache = new Dictionary<HttpClientCacheKey, Lazy<HttpClientCacheItem>>(GlobalConcurrentHttpClientCache);
+
+            return true;
+        }
+
+        return false;
+    }
+
+    public void Clear()
+    {
+        GlobalConcurrentHttpClientCache.Clear();
+        GlobalHttpClientCache = new Dictionary<HttpClientCacheKey, Lazy<HttpClientCacheItem>>();
+    }
+
+    private static HttpClient CreateClient(HttpClientCacheKey key, Func<HttpClientHandler, HttpClient> createHttpClient)
+    {
+        var httpMessageHandler = CreateHttpMessageHandler(key.Certificate,
+            setSslProtocols: true,
+            useCompression: key.UseCompression,
+            hasExplicitlySetCompressionUsage: key.HasExplicitlySetCompressionUsage,
+            key.PooledConnectionLifetime,
+            key.PooledConnectionIdleTimeout
+        );
+
+        var httpClient = createHttpClient(httpMessageHandler);
+
+        ConfigureHttpClient(httpClient);
+
+        return httpClient;
+    }
+
+    internal static void ConfigureHttpClient(HttpClient httpClient)
+    {
+        httpClient.Timeout = RequestExecutor.GlobalHttpClientTimeout;
+    }
+
+    internal static HttpClientHandler CreateHttpMessageHandler(X509Certificate2 certificate, bool setSslProtocols, bool useCompression, bool hasExplicitlySetCompressionUsage = false, TimeSpan? pooledConnectionLifetime = null, TimeSpan? pooledConnectionIdleTimeout = null)
+    {
+        var httpMessageHandler = new HttpClientHandler();
+
+        ConfigureHttpMessageHandler(httpMessageHandler, certificate, setSslProtocols, useCompression, hasExplicitlySetCompressionUsage, pooledConnectionLifetime, pooledConnectionIdleTimeout);
+
+        return httpMessageHandler;
+    }
+
+    internal static void ConfigureHttpMessageHandler(HttpClientHandler httpMessageHandler, X509Certificate2 certificate, bool setSslProtocols, bool useCompression, bool hasExplicitlySetCompressionUsage = false, TimeSpan? pooledConnectionLifetime = null, TimeSpan? pooledConnectionIdleTimeout = null)
```

Comment:
I wonder if we should abstract this method as well (with a parameters class instead the args)
This might allow us to avoid the many `#if NETCOREAPP3_1_OR_GREATER`/`PlatformNotSupportedException`  and fine tune for client or server handler

Reply:
Do not see benefit of that at the moment, it is internal